pkg/image/importer/importer.go

```diff
@@ -372,27 +392,34 @@ func importRepositoryFromDocker(ctx gocontext.Context, retriever RepositoryRetri
 			continue
 		}
 		limiter.Accept()
-		m, err := s.Get(d)
+		manifest, err := s.Get(ctx, d)
 		if err != nil {
 			glog.V(5).Infof("unable to access digest %q for repository %#v: %#v", d, repository, err)
-			switch {
-			case isDockerError(err, v2.ErrorCodeManifestUnknown):
-				ref := repository.Ref
-				ref.Tag, ref.ID = "", importDigest.Name
-				err = kapierrors.NewNotFound(api.Resource("dockerimage"), ref.Exact())
-			case isDockerError(err, errcode.ErrorCodeUnauthorized):
-				err = kapierrors.NewUnauthorized(fmt.Sprintf("you may not have access to the Docker image %q", repository.Ref.Exact()))
-			case strings.HasSuffix(err.Error(), "no basic auth credentials"):
-				err = kapierrors.NewUnauthorized(fmt.Sprintf("you may not have access to the Docker image %q", repository.Ref.Exact()))
+			importDigest.Err = formatRepositoryError(repository, "", importDigest.Name, err)
+			continue
+		}
+
+		if signedManifest, isSchema1 := manifest.(*schema1.SignedManifest); isSchema1 {
+			importDigest.Image, err = schema1ToImage(signedManifest, d)
+		} else if deserializedManifest, isSchema2 := manifest.(*schema2.DeserializedManifest); isSchema2 {
+			imageConfig, err := b.Get(ctx, deserializedManifest.Config.Digest)
+			if err != nil {
+				glog.V(5).Infof("unable to access to image config using digest %q for repository %#v: %#v", d, repository, err)
+				importDigest.Err = formatRepositoryError(repository, "", importDigest.Name, err)
```

Comment:
This will be matched by `v2.ErrorCodeManifestUnknown` case, which would return wrong error. We should return not found for docker image config.